stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/header/MessageListHeader.kt

```diff
@@ -86,55 +98,58 @@ public fun MessageListHeader(
             verticalAlignment = Alignment.CenterVertically,
         ) {
 
-            BackButton(
-                modifier = Modifier
-                    .size(36.dp),
-                imageVector = Icons.Default.ArrowBack,
-                onBackPressed = onBackPressed
-            )
+            leadingContent()
 
-            MessagesHeaderTitle(
-                modifier = Modifier
-                    .weight(6f)
-                    .fillMaxHeight()
-                    .clickable(
-                        interactionSource = remember { MutableInteractionSource() },
-                        indication = null,
-                        onClick = onHeaderActionClick
-                    ),
-                title = title,
-                subtitle = subtitle,
-                isNetworkAvailable = isNetworkAvailable
-            )
+            titleContent()
 
-            ChannelAvatar(
-                modifier = Modifier.size(36.dp),
-                channel = channel,
-                currentUser = currentUser,
-                contentDescription = channel.name
-            )
+            trailingContent()
         }
     }
 }
 
 /**
- * Wrapper for the title of the header, that handles if we should show a loading view for network,
+ * Default header title, that handles if we should show a loading view for network,
  * or the channel information.
  *
+ * @param channel The channel used for the title information.
  * @param modifier Modifier for styling.
- * @param title The title of the header.
- * @param subtitle The subtitle of the header - usually information about channel members.
+ * @param messageMode Currently active message mode, used to define the title information.
+ * @param onHeaderActionClick Handler for when the user taps on the header content.
  * @param isNetworkAvailable A flag that governs if we show the subtitle or the network loading view.
  */
 @Composable
-private fun MessagesHeaderTitle(
+public fun DefaultMessageHeaderTitle(
+    channel: Channel,
     modifier: Modifier,
-    title: String,
-    subtitle: String,
-    isNetworkAvailable: Boolean,
+    messageMode: MessageMode = Normal,
+    onHeaderActionClick: (Channel) -> Unit = {},
+    isNetworkAvailable: Boolean = true,
 ) {
+    val memberCount = channel.memberCount
+    val onlineCount = channel.members.count { it.user.online }
+
+    val channelName = channel.getDisplayName()
```

Comment:
We could probably inline this value to wherever it's used - I think it's never used more than once

Reply:
Channel name is used in two places, but the counts are not, so I can inline those two!

Reply:
Only one of those two usages will actually happen though (one is in the Normal, the other in the Thread case), so even if we inline it we'll only ever fetch the value once

Reply:
That's true - I had figued pulling it out is okay in the sense that if we want to customize the display name or anything it'll be easier to update it in one place.

But thinking about it, there's really not much need for that!